### DISCUSSION TITLE
Is there a simple way to set multiple folders as static directories?

### DISCUSSION THREAD

**Discussion by Aobanana-chan** [actix/actix-web#3398] [Q&A]
For some reasons, I am migrating my website from Node.js/Fastify to Actix Web, but I am having some difficulties setting up static directories. I tried to mount two directories, "./public" and "./dist" to "/", but it failed.
Here is a simplified example code:
```
App::new()
            .service(Files::new("/", "./dist"))
            .service(Files::new("/", "./public"))
```
In the first service, if the file is not found, it directly responds with a 404 instead of moving to the next service. I realize this approach doesn't work.

After roughly reading the actix_files code, I think I might be able to modify serve_from to be a Vec and change the relevant code to meet my needs. However, this would require copying the entire source code of actix_files, which might not be ideal. Therefore, I would like to know if there is a simpler way to achieve this functionality.

Thanks a lot.

**Comment by Aobanana-chan** [comment]
Finally, I decided to fork the repository and modify the implementation of actix_files. Now it can accept a form like `Files::new("/", vec!["./dist", "./public"])` to meet my needs. I am not sure if such a requirement is common, but if anyone is interested, they can use my forked version:
`actix-files = { git = "https://github.com/Aobanana-chan/actix-web.git", branch = "master" }`
Note that the `show_index `feature conflicts slightly with this requirement. If `show_index `is enabled and the number of directories provided is not 1, I have simply thrown a panic for this.

**Reply by robjtede** [reply]
Thank you for linking to your solution here. We could potentially use this as a base for future work. #3402
